### PR TITLE
First working version

### PR DESCRIPTION
**Pull request by avanier**
I wrote this cookbook because reasons.

### CODE REVIEW

**Comment by avanier** [comment]
@typositoire @brentonrails Would love a peer review if you have the time.

File: .kitchen.yml [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
   always_update_cookbooks: true
 
-verifier:
-  name: inspec
-
 platforms:
-  - name: ubuntu-16.04
   - name: centos-7.3
+    driver:
+      box: "centos/7"
```
Comment (Typositoire) [inline .kitchen.yml]: Booo for centos

File: README.md [hunk 2]
```diff
@@ -4,6 +4,29 @@ Installs and configures Concourse `web` and/or `worker`.
...
+Now, don't go complaining to Vito, I friggin' warned you.
+
+## Why should I run this cookbook?
+
+It would be appropriate for :
+
+  - Proof of concepts
+  - Debugging
+  - Unsupported shenanigans
+  - Trolling
```
Comment (Typositoire) [inline README.md]: +1 for trolling

File: attributes/default.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+default['concourse']['run_web'] = true
+default['concourse']['run_worker'] = true
+
+default['concourse']['local_postgres'] = true
+
+default['concourse']['web_data_bag'] = 'concourse/web-secrets'
+default['concourse']['worker_data_bag'] = 'concourse/worker-secrets'
+
+default['concourse']['worker']['CONCOURSE_BAGGAGECLAIM_DRIVER'] = 'btrfs'
```
Comment (Typositoire) [inline attributes/default.rb]: These should be in `worker.rb` attribute file

File: attributes/default.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+default['concourse']['worker']['CONCOURSE_BAGGAGECLAIM_VOLUMES'] = '/srv/concourse/volumes'
+default['concourse']['worker']['CONCOURSE_NAME'] = node['fqdn']
+default['concourse']['worker']['CONCOURSE_TSA_PUBLIC_KEY'] = '/etc/concourse/keys/web/public'
+default['concourse']['worker']['CONCOURSE_WORK_DIR'] = '/srv/concourse/work'
+default['concourse']['worker']['CONCOURSE_WORKER_PRIVATE_KEY'] = '/etc/concourse/keys/worker/private'
+
+default['concourse']['web']['CONCOURSE_BASIC_AUTH_PASSWORD'] = '___FULLFILLED_BY_DATABAG___'
```
Comment (Typositoire) [inline attributes/default.rb]: These should be in `web.rb` attribute file

File: attributes/default.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+default['concourse']['web']['CONCOURSE_BASIC_AUTH_USERNAME'] = '___FULLFILLED_BY_DATABAG___'
+default['concourse']['web']['CONCOURSE_POSTGRES_PASSWORD'] = nil
+default['concourse']['web']['CONCOURSE_POSTGRES_USER'] = 'postgres'
+default['concourse']['web']['CONCOURSE_TSA_AUTHORIZED_KEYS'] = '/etc/concourse/keys/worker/public'
+default['concourse']['web']['CONCOURSE_TSA_HOST_KEY'] = '/etc/concourse/keys/web/private'
+default['concourse']['web']['CONCOURSE_SESSION_SIGNING_KEY'] = '/etc/concourse/keys/web/session'
+
+default['postgresql']['enable_pgdg_yum'] = true
```
Comment (Typositoire) [inline attributes/default.rb]: These should be set in a recipe since you're overwriting an existing cookbook value

File: recipes/default.rb [hunk 6]
```diff
@@ -16,3 +16,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
```
Comment (Typositoire) [inline recipes/default.rb]: Add OS validation since you're assuming YUM repo at some place.

File: attributes/postgres.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+default['concourse']['local_postgres'] = true
+
+default['postgresql']['enable_pgdg_yum'] = true
```
Comment (Typositoire) [inline attributes/postgres.rb]: These should be set in the recipes since it's modifying an existing cookbook.

File: .kitchen.yml (hunk 1 above)
Comment (avanier) [reply .kitchen.yml]: CentOS all the things!

**Pull request merged**